README.md

```diff
@@ -453,6 +453,73 @@ As you can see all the classes in a class hierarchy actually share oneclass vari
   end
 ```
 
+- Avoid explicit use of instance variables
+
+```ruby
+  # bad
+  class SomeClass
+    def initialize(foo)
+      @foo = foo
+    end
+  end
+  
+  def bar
+    @foo == "Foo"
+  end
+  
+  # good
+  class SomeClass
+    attr_reader :foo
+    
+    def initialize(foo)
+      @foo = foo
+    end
+    
+    def bar
+      foo == "Foo"
+    end
+  end
+  
+  # better
+  class SomeClass
+    def initialize(foo)
+      @foo = foo
+    end
+    
+    def bar
+      foo == "Foo"
+    end
+    
+    private
+    
+    attr_reader :foo
+    
+  end
+```
+
+- Avoid complex logic in the initialiser
```

Comment:
I've moved towards preferring initialisation to only perform straight ivar assignment from arguments. I've been bitten too many times by hidden assumptions in initialisers, and you end up with a bunch of dependencies out the gate. If you want to do complex initialisation, you don't necessarily need a factory class, just a factory method that does the set up then hands the retrieved values to `new`. At least that way you're being explicit about what the factory does:

```ruby
class Foo
  def self.from_id(id)
    company = Company.find(id)
    new company
  end

  def initialize(company)
    @company = company
  end
end
```